docs/release.md

```diff
@@ -7,6 +7,7 @@ We release both core and contrib collectors with the same versions where the con
 1. The [Contrib](#releasing-opentelemetry-collector-contrib) collector.
 1. The [artifacts](#producing-the-artifacts)
 
+<!-- markdown-link-check-disable-next-line -->
```

Comment:
Why disabling this?

Reply:
For some reason the github link is returning a 403 error.  Rather than spend time trying to figure out why the github action is getting Forbidden errors when trying to access a github doc, I opted to ignore it.  We could add 403's in the to check_links_config's aliveStatusCodes, as 403 means its definitely a real link.

Reply:
Would like to discuss this in a separate PR since is the only controversial change.